editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedBeanPropertiesSection.java

```diff
@@ -0,0 +1,323 @@
+		IProject project = selectedEP.getCamelFile().getResource().getProject();
+		String propName = p.getName();
+		
+		if (propName.equals(CamelBean.PROP_CLASS)) {
+			txtFieldCreator = createTextFieldWithClassBrowseAndNew(p, page);
+			validator = new BeanClassExistsValidator(project);
+		} else if (propName.equals(CamelBean.PROP_INIT_METHOD)
+				|| propName.equals(CamelBean.PROP_DESTROY_METHOD)) {
+			txtFieldCreator = createTextFieldWithNoArgMethodBrowse(p, page);
+			validator = new PropertyMethodValidator(modelMap, project);
+		} else if (propName.equals(CamelBean.PROP_FACTORY_METHOD)
+				|| propName.equals(CamelBean.PROP_FACTORY_BEAN)) {
+			txtFieldCreator = createTextFieldWithMethodBrowse(p, page);
+			validator = new PropertyMethodValidator(modelMap, project);
+		} else if (propName.equals(CamelBean.PROP_ID)) {
+			txtFieldCreator = createTextField(p, page);
+			validator = new NewBeanIdPropertyValidator(p, selectedEP);
+		} else if (CamelComponentUtils.isTextProperty(p) || CamelComponentUtils.isCharProperty(p)) {
+			txtFieldCreator = createTextField(p, page);
+			if (p.getRequired() != null && "true".contentEquals(p.getRequired())) { //$NON-NLS-1$
+				validator = new PropertyRequiredValidator(p);
+			}
+		} else if (CamelComponentUtils.isUnsupportedProperty(p)) {
+			// handle unsupported props
+			new UnsupportedParameterPropertyUICreatorForAdvanced(dbc, modelMap, eip, selectedEP, p, page,
+					getWidgetFactory()).create();
+		}
+
+		if (txtFieldCreator != null) {
+			ISWTObservableValue uiObservable = handleObservable(txtFieldCreator, p);
+			if (uiObservable != null) {
+				bindField(validator, uiObservable, p);
+			}
+		}
+	}	
+	
+	private ISWTObservableValue handleObservable(AbstractTextFieldParameterPropertyUICreator txtFieldCreator, Parameter p) {
+		// initialize the map entry
+		modelMap.put(p.getName(), txtFieldCreator.getControl().getText());
+		// create observables for the control
+		return txtFieldCreator.getUiObservable();
+	}
+	
+	private void bindField(IValidator validator, ISWTObservableValue uiObservable, Parameter p) {
+		// create UpdateValueStrategy and assign to the binding
+		UpdateValueStrategy strategy = new UpdateValueStrategy();
+		strategy.setBeforeSetValidator(validator);
+
+		// create observables for the Map entries
+		IObservableValue<Object> modelObservable = Observables.observeMapEntry(modelMap, p.getName());
+		// bind the observables
+		Binding bindValue = dbc.bindValue(uiObservable, modelObservable, strategy, null);
+		ControlDecorationSupport.create(bindValue, SWT.TOP | SWT.LEFT);
+	}
+	
+	/*
+	 * (non-Javadoc)
+	 * 
+	 * @see org.eclipse.ui.views.properties.tabbed.AbstractPropertySection#
+	 * createControls (org.eclipse.swt.widgets.Composite,
+	 * org.eclipse.ui.views.properties.tabbed.TabbedPropertySheetPage)
+	 */
+	@Override
+	public void createControls(Composite parent, TabbedPropertySheetPage aTabbedPropertySheetPage) {
+		this.toolkit = new FormToolkit(parent.getDisplay());
```

Comment:
apart from the test on the form, this method is a perfect duplicate of the same method in the sibling class https://github.com/jbosstools/jbosstools-fuse/blob/master/editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/properties/AdvancedEndpointPropertiesSection.java#L302
what do you think to factorize it?